File: java-maven/src/main/java/javamaven/javamaven/entity/Holiday.java
package javamaven.javamaven.entity;

import javax.persistence.*;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;

@Entity
@Table(name = "holiday")
public class Holiday {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "date")
    private String date;

    @Transient
    private String day;

    public Holiday() {
        this.name = "";
        this.date = "";
    }

    public Holiday(String name, String date) {
        this.name = name;
        this.date = date;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getDay() {
        try {
            SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd");
            Date _date = df.parse(this.date);
            DateFormat df2 = new SimpleDateFormat("EEEE");
            return df2.format(_date);
        } catch (java.text.ParseException e) {
            e.printStackTrace();
            return "Unable to parse date";
        }
    }

    public void setDay(String day) {
        this.day = day;
    }

    @Override
    public String toString() {
        return "Holiday{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", date='" + date + '\'' +
                ", day='" + day + '\'' +
                '}';
    }
}

File: kotlin-maven/src/main/kotlin/kotlinholidaytracker/kotlinholidaytracker/entity/Holiday.kt
package kotlinholidaytracker.kotlinholidaytracker.entity

import java.text.DateFormat
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "holiday")
data class Holiday (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    var id: Long?,

    @Column(name = "name")
    var name: String?,

    @Column(name = "date")
    var date: String?,
) {
    val day: String
        get() {
            return try {
                val df = SimpleDateFormat("yyyy-MM-dd")
                val date2 = df.parse(this.date)
                val df2: DateFormat = SimpleDateFormat("EEEE")
                df2.format(date2)
            } catch (e: ParseException) {
                e.printStackTrace()
                "Unable to parse date"
            }
        }
}